### ISSUE TITLE
A suggestion on the storage path of generated header file

### ISSUE THREAD

**Issue by Cheney-W** [finos/OpenMAMA#448]
I'm a member of Microsoft VCPKG team.
In our internal test, port **workflow** installation failed due to it used the list.h file which generated by OpenMAMA.
I have added a patch to workaround this issue, please see the PR: https://github.com/microsoft/vcpkg/pull/16388.
If possible, I think integrating the generated header file into a folder which named with port name will avoid the problem of duplicate file names.

**Comment by fquinner** [comment]
Thanks - yes unfortunately this is a hangover from the days of being a non-open library when our stuff would be installed in an entirely separate directory so things like generically named headers weren't an issue... It's something that we wanted to fix properly in our next major release but realistically we wouldn't get around to in the short to medium term.

The issue with this change is that (if I have read it correctly), everyone who currently builds against this library would need to add "MAMA_ROOT/openmama" to their include path.

I wonder if a better solution may to allow (at build time) a cached CMAKE_INSTALL_INCLUDE_DIRECTORY variable, then some vcpkg magic will take care of automatically including *that* include directory when building against it. That way we could preserve existing layouts (and `#include <mama/whatever.h>` will continue to work. I haven't messed with vcpkg in a while let me pull down what you have done and see what the impact would be etc. Thanks for bring it up!

**Comment by fquinner** [comment]
I have raised https://github.com/Cheney-W/vcpkg/pull/1 with an alternative suggestion please let me know what you think (I raised against your existing PR branch so it will get added to your existing PR if you think it's OK).

**Comment by Cheney-W** [comment]
Thanks for your reply! 
I saw that the public header files will be included by some of the header files of openmama itself, such as:
`list.h -> include\mama\integration\transport.h`
`list.h -> include\mama\integration\types.h`
`property.h -> include\mama\integration\mama.h`
`wlock.h -> include\wombat\MRSWLock.h`
So will the current changes affect them?
And if we implement this patch, will upstream make some corresponding changes?

**Comment by Cheney-W** [comment]
@fquinner 
Do you have any update?

**Comment by Cheney-W** [comment]
@fquinner 
Because of some unexpected operation, I remove the PR: https://github.com/microsoft/vcpkg/pull/16388.
And I submitted a new PR: https://github.com/microsoft/vcpkg/pull/16725 , this PR is made according to your suggestion. Could you please help review it?

**Comment by fquinner** [comment]
@Cheney-W Yes this looks good to me. The fact is most users won't actually be using those headers anyway and in future I'll try and get these moved into a subdirectory by default. When that time comes I'll update this workaround in VCPKG but for now yes this looks perfectly fine to me.

**Comment by fquinner** [comment]
Succeeded by https://github.com/finos/OpenMAMA/issues/454
